From 9afedd6c330eba4bdf8b55754156b2d39c45ea19 Mon Sep 17 00:00:00 2001
From: mrapple <tony@oc.tc>
Date: Sun, 16 Jun 2013 16:57:59 -0500
Subject: [PATCH] Add getting and setting of entity count


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index a90b390..cc1100e 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -707,4 +707,18 @@ public final class Bukkit {
     public static UnsafeValues getUnsafe() {
         return server.getUnsafe();
     }
+
+    /**
+     * @see Server#getEntityCount()
+     */
+    public static int getEntityCount() {
+        return server.getEntityCount();
+    }
+
+    /**
+     * @see Server#setEntityCount()
+     */
+    public static void setEntityCount(int count) {
+        server.setEntityCount(count);
+    }
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 3b6ea63..ff8cf32 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -813,4 +813,18 @@ public interface Server extends PluginMessageRecipient {
      */
     @Deprecated
     UnsafeValues getUnsafe();
+
+    /**
+     * Gets the current entity count
+     *
+     * @return current entity count
+     */
+    int getEntityCount();
+
+    /**
+     * Sets the current entity count
+     *
+     * @param count The new entity count
+     */
+    void setEntityCount(int count);
 }
-- 
1.8.5.2.msysgit.0

