From d218d0d7a6cec173f75541b1a54502dced96f086 Mon Sep 17 00:00:00 2001
From: mrapple <tony@oc.tc>
Date: Wed, 29 May 2013 23:52:37 -0500
Subject: [PATCH] Visibility API


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index e18eedd..c0e2feb 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -845,6 +845,8 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
      */
     public void hidePlayer(Player player);
 
+    public void hide(Entity entity);
+
     /**
      * Allows this player to see a player that was previously hidden
      *
@@ -852,8 +854,19 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
      */
     public void showPlayer(Player player);
 
+    public void show(Entity entity);
+
+    /**
+     * Checks to see if an entity has been hidden from this player
+     *
+     * @param entity Player to check
+     * @return True if the provided entity is not being hidden from this
+     *     player
+     */
+    public boolean canSee(Entity entity);
+
     /**
-     * Checks to see if a player has been hidden from this player
+     * Checks to see if an entity has been hidden from this player
      *
      * @param player Player to check
      * @return True if the provided player is not being hidden from this
-- 
1.8.5.2.msysgit.0

