From 8d0e79e2aae3c4a40c8e3d8b52f798bc8e4a52b1 Mon Sep 17 00:00:00 2001
From: Marcos Vives Del Sol <socram8888@gmail.com>
Date: Sun, 28 Jul 2013 21:56:56 +0200
Subject: [PATCH] Add getHostname to Player


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index c0e2feb..20167ae 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -1078,4 +1078,14 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
     public void removeIgnorantEffect(PotionEffectType effect);
 
     public boolean hasIgnorantEffect(PotionEffectType effect);
+
+    /**
+     * Gets the hostname (an IP or a DNS domain) and port the client has used
+     * to connect to this server. This is generated client-side, and it may
+     * return a blank string if the client didn't send it, or an invalid
+     * hostname, as the server doesn't check it.
+     *
+     * @return The hostname and port
+     */
+    public String getHostname();
  }
-- 
1.8.5.2.msysgit.0

